ui/ozone/platform/wayland/wayland_keyboard.cc

```diff
@@ -97,32 +100,19 @@ void WaylandKeyboard::Key(void* data,
                           uint32_t key,
                           uint32_t state) {
   WaylandKeyboard* keyboard = static_cast<WaylandKeyboard*>(data);
-  keyboard->connection_->set_serial(serial);
+  DCHECK(keyboard);
 
-  DomCode dom_code =
-      KeycodeConverter::NativeKeycodeToDomCode(key + kXkbKeycodeOffset);
-  if (dom_code == ui::DomCode::NONE)
-    return;
-
-  uint8_t flags = keyboard->event_modifiers_.GetModifierFlags();
-  DomKey dom_key;
-  KeyboardCode key_code;
-  if (!KeyboardLayoutEngineManager::GetKeyboardLayoutEngine()->Lookup(
-          dom_code, flags, &dom_key, &key_code))
-    return;
+  keyboard->connection_->set_serial(serial);
 
   bool down = state == WL_KEYBOARD_KEY_STATE_PRESSED;
+  int device_id = keyboard->obj_.id();
 
-  // TODO(tonikitoo,msisov): only the two lines below if not handling repeat.
-  int flag = ModifierDomKeyToEventFlag(dom_key);
-  keyboard->UpdateModifier(flag, down);
+  keyboard->auto_repeat_handler_.UpdateKeyRepeat(
+      key, down, false /*suppress_auto_repeat*/, device_id);
 
-  ui::KeyEvent event(
-      down ? ET_KEY_PRESSED : ET_KEY_RELEASED, key_code, dom_code,
-      keyboard->event_modifiers_.GetModifierFlags(), dom_key,
-      base::TimeTicks() + base::TimeDelta::FromMilliseconds(time));
-  event.set_source_device_id(keyboard->obj_.id());
-  keyboard->callback_.Run(&event);
+  keyboard->DispatchKeyEvent(
+      key, down, false /*repeat*/,
+      base::TimeTicks() + base::TimeDelta::FromMilliseconds(time), device_id);
```

Comment:
So, you decided to use time passed by Wayland, didn't you? Before that you switched to EventTimeForNow, IIRC.

Reply:
Right. As part of this patch, I decided not to change TimeDelta -> EventTimeForNow, since it is unrelated. It just keeps things as it is.

Lets try to consolidate the use of one of another in a secondary CL.